src/modules/usart.rs

```diff
@@ -0,0 +1,258 @@
+use core::prelude::*;
+use core::default::Default;
+
+use collections::vec::Vec;
+use collections::string::String;
+
+use {cmu, gpio, usart};
+use gpio::Port;
+
+use self::Location::*;
+
+type PortPin = Option<(Port, u32)>;
+
+/// Possible locations for the different Usart's. Not all locations is available for every Usart.
+#[derive(Copy, Debug)]
+pub enum Location {
+    Loc0,
+    Loc1,
+    Loc2,
+    Loc3,
+    Loc4,
+    Loc5,
+}
+
+/// Possible different Usart's to configure.
+#[derive(Copy, Debug)]
+pub enum Config {
+    Usart0(Location),
+    Usart1(Location),
+    Usart2(Location),
+}
+
+#[allow(dead_code)]
+pub struct Usart {
+    config: Config,
+    baudrate: u32,
+    usart: &'static mut usart::Usart,
+    line_break: &'static str,
+}
+
+/// Initializes and returns a new Usart1 instance with Location 1 and a baudrate of 9600.
+impl Default for Usart {
+    fn default() -> Usart {
+        let mut usart = Usart {
+            config: Config::Usart1(Loc1),
+            baudrate: 9600,
+            usart: usart::Usart::usart1(),
+            line_break: "\0\r",
+        };
+
+        usart.init();
+        usart
+    }
+}
+
+impl Usart {
+    /// Returns a new Usart instance. The Usart have to be manually initialized and it is
+    /// configured to return Strings when it receives either a `\0` or a `\r` character.
+    pub fn new(config: Config, baudrate: u32) -> Usart {
+        Usart {
+            config: config,
+            baudrate: baudrate,
+            usart: match config {
+                Config::Usart0(_) => usart::Usart::usart0(),
+                Config::Usart1(_) => usart::Usart::usart1(),
+                Config::Usart2(_) => usart::Usart::usart2(),
+            },
+            line_break: "\0\r",
+        }
+    }
+
+    /// Called in order to initialize the Usart according to it's config
+    pub fn init(&mut self) {
+        // Enable clock for HF peripherals
+        cmu::clock_select_set(cmu::Clock::HF, cmu::Select::HFRCO);
+
+        // Configure usart
+        let (usart_clock, usart) = match self.config {
+            Config::Usart0(_) => (cmu::Clock::USART0, usart::Usart::usart0()),
+            Config::Usart1(_) => (cmu::Clock::USART1, usart::Usart::usart1()),
+            Config::Usart2(_) => (cmu::Clock::USART2, usart::Usart::usart2()),
+        };
+        self.usart = usart;
+        cmu::clock_enable(usart_clock, true);
+        self.usart.init_async(&usart::InitAsync {
```

Comment:
Why not? 
This is the only one we've used so far, and a synchronous one would require a clock, right?